Modules/@babylonjs/react-native-iosandroid/android/src/main/java/com/babylonreactnative/EngineView.java

```diff
@@ -67,34 +70,58 @@ public void surfaceDestroyed(SurfaceHolder holder) {
     public void setAntiAliasing(Integer value) {
         BabylonNativeInterop.updateMSAA(value);
     }
+
+    public void setTopMost(Boolean topMost) {
+        setIsTransparentAndTopMost(this.isTransparent, topMost);
+    }
     // ------------------------------------
     // TextureView related
-
     public void setIsTransparent(Boolean isTransparent) {
+        setIsTransparentAndTopMost(isTransparent, this.topMost);
+    }
+
+    public void setIsTransparentAndTopMost(Boolean isTransparent, Boolean topMost) {
+        if (this.surfaceView != null) {
+            this.surfaceView.setVisibility(View.GONE);
+            this.surfaceView = null;
+        }
+        if (this.textureView != null) {
+            this.textureView.setVisibility(View.GONE);
+            this.textureView = null;
+        }
         if (isTransparent) {
```

Comment:
These conditions seems like they could be simplified. Conceptually can't it just be:
```ts
if (isTransparent && !topMost) {
  // use TextureView
} else {
  // use SurfaceView (configured according to isTransparent and topMost)
}
```

It seems like this would simplify the logic and remove some code duplication, which should also reduce opportunities for coding errors. Speaking of which, isn't the first case of creating the `SurfaceView` missing the `addCallback` which deals with resizing?

Reply:
Indeed! It's better.